auth/jwt/middleware.go

```diff
@@ -0,0 +1,106 @@
+package jwt
+
+import (
+	"errors"
+
+	"golang.org/x/net/context"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/go-kit/kit/endpoint"
+)
+
+type contextKey string
+
+const (
+	// JWTTokenContextKey holds the key used to store a JWT Token in the context
+	JWTTokenContextKey contextKey = "JWTToken"
+	// JWTClaimsContxtKey holds the key used to store the JWT Claims in the context
+	JWTClaimsContextKey contextKey = "JWTClaims"
+)
+
+var (
+	ErrTokenContextMissing     = errors.New("Token up for parsing was not passed through the context")
+	ErrTokenInvalid            = errors.New("JWT Token was invalid")
+	ErrTokenExpired            = errors.New("JWT Token is expired")
+	ErrTokenMalformed          = errors.New("JWT Token is malformed")
+	ErrTokenNotActive          = errors.New("Token is not valid yet")
+	ErrUnexpectedSigningMethod = errors.New("Unexpected signing method")
+)
+
+type Claims map[string]interface{}
+
+// Create a new JWT token generating middleware, specifying signing method and the claims
+// you would like it to contain. Particularly useful for clients.
+func NewSigner(kid string, key []byte, method jwt.SigningMethod, claims Claims) endpoint.Middleware {
```

Comment:
Comment should be of the form "NewSigner creates a new...". Also, can you mention what kid is?